app/repositories/clans.py

```diff
@@ -106,32 +126,32 @@ async def fetch_many(
         params["limit"] = page_size
         params["offset"] = (page - 1) * page_size
 
-    recs = await app.state.services.database.fetch_all(query, params)
-    return [dict(rec) for rec in recs]
+    clan = await app.state.services.database.fetch_all(query, params)
+    return cast(list[Clan], clan) if clan is not None else None

```

Comment:
```suggestion
    clans = await app.state.services.database.fetch_all(query, params)
    return cast(list[Clan], clans) if clans is not None else None

```

Reply:
actually, can `fetch_all` even return `None?` or does it always return `list[T]`?